### PR TITLE
JSON appender script

### PR DESCRIPTION
**Pull request by niccolopaganini**
- python script to append anything to a JSON file in any given directory (configs in this case)
- `environment.yml` file for the aforementioned python script

### CODE REVIEW

**Comment by niccolopaganini** [comment]
issue here: https://github.com/e-mission/op-admin-dashboard/issues/75 

Added a template for the time being

File: bin/environment.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name: myenv
```
Comment (shankari) [inline bin/environment.yml:1]: nit: change this to something more meaningful

File: bin/kv_pair_appender.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        file_path = os.path.join(dir_path, i)
+
+        with open(file_path, "r") as file: 
+            try:
+                json_data = json.load(file)
+            except json.JSONDecodeError:
+                print(f"Error parsing JSON in file: {file_path}")
+                continue 
+
+        json_data.update(dd_app)
```
Comment (shankari) [inline bin/kv_pair_appender.py:29]: you are going to run into an issue with this when you test because the newly added fields are not at the top level

**Review by shankari — CHANGES_REQUESTED** [review]

File: bin/environment.yml (hunk 1 above)
Comment (niccolopaganini) [reply bin/environment.yml:1]: name changed to `kv_pair_appender_env`

File: bin/kv_pair_appender.py (hunk 2 above)
Comment (niccolopaganini) [reply bin/kv_pair_appender.py:29]: If I understand it correctly, it means that the `data_trajectories_columns_exclude ` should be under `admin_dashboard`
```
 "admin_dashboard": {
        "overview_users": true,
        "overview_active_users": true,
        "overview_trips": true,
        "overview_signup_trends": true,
        "overview_trips_trend": true,
        "data_uuids": true,
        "data_trips": true,
        "data_trips_columns_exclude": ["data.start_loc.coordinates", "data.end_loc.coordinates"],
        "additional_trip_columns": [],
        "data_uuids_columns_exclude": [],
        "token_generate": false,
        "token_prefix": "nrelop",
        "map_heatmap": true,
        "map_bubble": true,
        "map_trip_lines": false,
        "push_send": false,
        "options_uuids": true,
        "options_emails": true
    }
```

**Comment by niccolopaganini** [comment]
@shankari please let me know if I understood what you said [#70](https://github.com/e-mission/op-admin-dashboard/issues/70#issuecomment-1744049354) here correctly.

**Comment by shankari** [comment]
@niccolopaganini please see my response. I don't want all the files to be changed.
Also, as you can see, this code has introduced several extraneous changes. The only changes to the files should be the ones that want. You probably want to write in UTF-8 format with appropriate indentation to avoid extraneous changes.

**Comment by shankari** [comment]
At this point, you may want to submit a separate PR by making the changes manually so we can get this done today, and work on cleaning up the script later.

**Comment by niccolopaganini** [comment]
checklist:
1. encoding format ✅ 
2. only target files ✅ 
3. see if it has any "side effects"

**Comment by niccolopaganini** [comment]
troubleshooted for like the last 25 minutes and had to force-push the changes. revert all the config files that don't need any change. Changes only applied by script to the JSON files that need change. Hopefully, this meets all the criteria.

**Comment by shankari** [comment]
@niccolopaganini this is much better - the files are in UTF-8.
However, as I said in https://github.com/e-mission/op-admin-dashboard/issues/70#issuecomment-1744049354

> The list of programs that should **_not_** exclude the user_id is:

Instead, you **have** excluded the user_id from just those programs. You need to flip this around.

There are still extraneous formatting changes.

As I already said:

> The only changes to the files should be the ones that we want. 

This means that the only change should be the line for `data_trajectories_columns_exclude`. Everything else should be unchanged. Otherwise this is hard to review, and git blame gets too messed up. If you can't figure out the JSON dump arguments to enable this, feel free to use `jq` to edit the files directly instead. If there is a reason why this doesn't work, **please document it**

**Comment by niccolopaganini** [comment]
I was thinking about how to tackle this problem and I found this to be a possible path:
- Work on/ Write a JSON Schema Validation script something like 
```
{
  k: val
  k: val
   {
     k: val
     k: val
...
   }
}
```
- Validate them
```
import jsonschema
# open schema
...
# iterate through the JSON files and then validate them
```
And these can go on an exception list or have their own schema:
> user_id is: ca-ebike, stage-program, stage-study, uprm-civic, uprm-nicr, usaid-laos-ev, uue, washingtoncommons.

I went through the actions in the repo to peek at what's happening there (I wanted to see if there is anything relatable but I couldn't). Is this a path I could take that would be fruitful @shankari?

**Comment by niccolopaganini** [comment]
For the current changes that I have rewritten, these are the errors I am getting:
```
ERROR in /Users/nseptank/e-mission-phone-nvsr-iter1/www/js/control/emailService.tsx
./www/js/control/emailService.tsx 5:4-5
[tsl] ERROR in /Users/nseptank/e-mission-phone-nvsr-iter1/www/js/control/emailService.tsx(5,5)
      TS1128: Declaration or statement expected.
ts-loader-default_8025a27c15fa23e2
 @ ./www/index.js 27:0-39

ERROR in /Users/nseptank/e-mission-phone-nvsr-iter1/www/js/control/emailService.tsx
./www/js/control/emailService.tsx 66:33-40
[tsl] ERROR in /Users/nseptank/e-mission-phone-nvsr-iter1/www/js/control/emailService.tsx(66,34)
      TS1005: ';' expected.
ts-loader-default_8025a27c15fa23e2
 @ ./www/index.js 27:0-39
```

**Pull request closed without merging**